app/controllers/accounts/users_controller.rb

```diff
@@ -21,6 +21,9 @@ def set_user
     end
 
     def role_params
-      { role: params.require(:user)[:role].presence_in(%w[ member administrator ]) || "member" }
+      {
+        role: params.require(:user)[:role].presence_in(%w[ member administrator ]) || "member",
+        banned: params.require(:user)[:banned] == "1"
```

Comment:
You don't have to do the comparison by hand, if the column is of type boolean, ActiveRecord will do the type casting for you - e.g. `1`, `true`, `TRUE` would all set the boolean to `true`.

I'm a bit unsure about the naming here, banning someone isn't really a role change?

Reply:
In my humble opinion banning is just an attribute of the user. The role did not change. So if in the future we will have more than 2 roles, you will be able to ban the user and keep its role unchanged, to simplify unbanning.

Reply:
There is a [with_defaults](https://api.rubyonrails.org/classes/ActionController/Parameters.html#method-i-with_defaults) method that you can use to set banned to fals if it's not passed.

> In my humble opinion banning is just an attribute of the user. The role did not change.

I fully agree, that's why I'm bringing this up - the change to allow the banned flag was to the `role_params` method. And banning someone isn't a role change, so I believe it should be its own method.